lib/file.coffee

```diff
@@ -67,3 +67,6 @@ class File
 
   isPathEqual: (pathToCompare) ->
     @path is pathToCompare or @realPath is pathToCompare
+
+  # Return a snapshot of the instance's filesystem properties for API consumption
+  getMetadata: -> {@symlink, @realPath, @status, @stats}
```

Comment:
I'm aware it's not 100% consistent, but how about storing a reference to the buffer's `File` instance on the second argument? `stats` won't be available, but the source file can be keyed to `bufferFile` or just `file`.

```
tree-view: { symlink, realPath, status, stats }
tabs: { symlink, realPath, sourceFile }
```

To be honest, I think we can probably justify having an extra filesystem call to return the missing info to stay consistent with `tree-view`'s handling. Let's face it: there won't be nearly as many icon-displaying tabs as there will be icon-displaying file listings.